### PR TITLE
Feature/fix pytket update

### PR DESCRIPTION
**Pull request by cqc-melf**

### CODE REVIEW

File: pytket/qir/conversion/conversion.py [hunk 1]
```diff
@@ -917,11 +927,15 @@ def condition_block() -> None:
                     # classical ops acting on registers returning bit
                     ssa_left = cast(  # type: ignore
                         Value,
-                        self._get_ssa_from_cl_reg_op(op.get_exp().args[0], module),
+                        self._get_ssa_from_cl_reg_op(
```
Comment (cqc-melf) [inline pytket/qir/conversion/conversion.py:923]: Mypy is not able to see that the type is in `_TK_CLOPS_TO_PYQIR_REG_BOOL`

File: pytket/qir/conversion/conversion.py [hunk 2]
```diff
@@ -499,10 +499,12 @@ def _get_ssa_from_cl_reg_op(
         self, reg: Union[BitRegister, RegAnd, RegOr, RegXor], module: tketqirModule
     ) -> Value:
         if type(reg) in _TK_CLOPS_TO_PYQIR_REG:
-            assert len(reg.args) == 2
+            assert len(reg.args) == 2  # type: ignore
```
Comment (cqc-melf) [inline pytket/qir/conversion/conversion.py:495]: Mypy is not able to see that the type is in `_TK_CLOPS_TO_PYQIR_REG`

File: pytket/qir/conversion/conversion.py [hunk 3]
```diff
@@ -499,10 +499,12 @@ def _get_ssa_from_cl_reg_op(
...
 
-            ssa_left = self._get_ssa_from_cl_reg_op(reg.args[0], module)
-            ssa_right = self._get_ssa_from_cl_reg_op(reg.args[1], module)
+            ssa_left = self._get_ssa_from_cl_reg_op(reg.args[0], module)  # type: ignore
```
Comment (cqc-melf) [inline pytket/qir/conversion/conversion.py:497]: Mypy is not able to see that the type is in `_TK_CLOPS_TO_PYQIR_REG`

File: pytket/qir/conversion/conversion.py [hunk 4]
```diff
@@ -883,11 +885,15 @@ def condition_block() -> None:
                     # classical ops acting on registers returning register
                     ssa_left = cast(  # type: ignore
                         Value,
-                        self._get_ssa_from_cl_reg_op(op.get_exp().args[0], module),
+                        self._get_ssa_from_cl_reg_op(
```
Comment (cqc-melf) [inline pytket/qir/conversion/conversion.py:881]: Mypy is not able to see that the type is in `_TK_CLOPS_TO_PYQIR_REG`

File: pytket/qir/conversion/conversion.py [hunk 5]
```diff
@@ -899,11 +905,15 @@ def condition_block() -> None:
                     # classical ops acting on bits returning bit
                     ssa_left = cast(  # type: ignore
                         Value,
-                        self._get_ssa_from_cl_bit_op(op.get_exp().args[0], module),
+                        self._get_ssa_from_cl_bit_op(
```
Comment (cqc-melf) [inline pytket/qir/conversion/conversion.py:901]: Mypy is not able to see that the type is in `_TK_CLOPS_TO_PYQIR_BIT`

File: pytket/qir/conversion/conversion.py [hunk 6]
```diff
@@ -899,11 +905,15 @@ def condition_block() -> None:
...
+                            op.get_exp().args[0], module  # type: ignore
+                        ),
                     )
                     ssa_right = cast(  # type: ignore
                         Value,
-                        self._get_ssa_from_cl_bit_op(op.get_exp().args[1], module),
+                        self._get_ssa_from_cl_bit_op(
```
Comment (cqc-melf) [inline pytket/qir/conversion/conversion.py:907]: Mypy is not able to see that the type is in `_TK_CLOPS_TO_PYQIR_BIT`

File: tests/conditional_test.py [hunk 7]
```diff
@@ -36,9 +36,9 @@ def test_pytket_qir_conditional() -> None:
     c = circ.add_c_register("c", 5)
     d = circ.add_c_register("d", 5)
     circ.H(0)
-    circ.add_classicalexpbox_register(a | b, c)
-    circ.add_classicalexpbox_register(c | b, d)
-    circ.add_classicalexpbox_register(c | b, d, condition=a[4])
+    circ.add_classicalexpbox_register(a | b, c)  # type: ignore
```
Comment (cqc-melf) [inline tests/conditional_test.py:39]: Mypy does not like this anymore, it only wants list of bits here, but I would like to keep it like this (same below)

File: pytket/qir/conversion/conversion.py [hunk 8]
```diff
@@ -424,11 +424,11 @@ def _get_optype_and_params(self, op: Op) -> tuple[OpType, Sequence[float]]:
         params: list = []
         if optype == OpType.ExplicitPredicate:
             if op.get_name() == "AND":
-                optype = BitWiseOp.AND
+                optype = BitWiseOp.AND  # type: ignore
```
Comment (cqc-melf) [inline pytket/qir/conversion/conversion.py]: Mypy is not able to see that BitWiseOp.XOR is an optype.

**Comment by cqc-melf** [comment]
Had some mypy issues, if you have any questions, please let me know

File: setup.py [hunk 9]
```diff
@@ -45,7 +45,7 @@
     packages=find_namespace_packages(include=["pytket.*"]),
     include_package_data=True,
     install_requires=[
-        "pytket == 1.19.0rc0",
+        "pytket == 1.20.0rc1",
```
Comment (cqc-alec) [inline setup.py:48]: 1.19.1 is more recent if you want to use the latest.

File: pytket/qir/conversion/conversion.py (hunk 8 above)
Comment (cqc-alec) [reply pytket/qir/conversion/conversion.py]: But it is actually a different enum, is this variable maybe mistyped?

**Review by cqc-alec — COMMENTED** [review]
I suggest adding `warn_unused_ignores = True` to `mypy.ini` as well.

File: setup.py (hunk 9 above)
Comment (cqc-melf) [reply setup.py:48]: The barrier change is not in 1.19.1

Comment (cqc-alec) [reply setup.py:48]: Oh I thought it was but you are right.

File: pytket/qir/conversion/conversion.py [hunk 10]
```diff
@@ -422,14 +422,7 @@ def _rebase_op_to_gateset(self, op: Op, args: list) -> Optional[Circuit]:
     def _get_optype_and_params(self, op: Op) -> tuple[OpType, Sequence[float]]:
         optype: OpType = op.type
         params: list = []
-        if optype == OpType.ExplicitPredicate:
```
Comment (cqc-melf) [inline pytket/qir/conversion/conversion.py:425]: I am not sure sure why this was in here, I was note able to construct a case where this was used. Everything is still working without this.

**Review by cqc-alec — APPROVED** [review]

**Pull request merged**